plugins/dac_data_manager.c

```diff
@@ -1237,23 +1227,46 @@ static void gui_manager_create(struct dac_data_manager *manager)
 	gtk_widget_show(hbox);
 }
 
+/* Uses the "TX*_I|Q_F* name convention to build a name for the channel at the given index.
+ * E.g. TX1_I_F1 - 0 (the index)
+ *      TX1_I_F2 - 1
+ *      TX1_Q_F1 - 2
+ *      TX1_Q_F2 - 3
+ *      TX2_I_F1 - 4
+ *      TX2_I_F2 - 5
+ *      ...
+ * Returns a newlly alocated char array.
+ */
+static char *build_default_channel_name_from_index(guint ch_index)
+{
+	guint tx_index = (ch_index / TX_NB_TONES) + 1; /* TX couting starts from 1 (not 0) */
+	guint tone_index = (ch_index % 2) + 1; /* There are always 2 tones (I/Q). Count starts at 1 */
+	const char i_q_type = (ch_index & 0x02) ? Q_CHANNEL : I_CHANNEL; /* First two indexes are I, next two are Q and so on*/
+
+	return g_strdup_printf("TX%i_%c_F%i", tx_index, i_q_type, tone_index);
+}
+
 #define TONE_ID "altvoltage"
 #define TONE_ID_SIZE (sizeof(TONE_ID) - 1)
 
-static const char * get_tone_name(struct iio_channel *ch)
+/* Returns a newlly allocated tone name for the given channel in case of
+ * success and NULL otherwise.
+ */
+static char * get_tone_name(struct iio_channel *ch)
 {
-	const char *name;
+	char *name;
 	char tone_index;
 
-	name = iio_channel_get_name(ch);
+	name = g_strdup( iio_channel_get_name(ch));
```

Comment:
does this really need to be `strdup-ed` ?

Reply:
Yes, because name can take an allocated array as well (from build_default_channel_name_from_index() ).